arangod/VocBase/Methods/Collections.cpp

```diff
@@ -518,14 +512,11 @@ Result Collections::properties(Context& ctxt, VPackBuilder& builder) {
 
 Result Collections::updateProperties(LogicalCollection& collection,
                                      velocypack::Slice const& props, bool partialUpdate) {
-  ExecContext const* exec = ExecContext::CURRENT;
+  ExecContext const& exec = ExecContext::current();
+  bool canModify = exec.canUseCollection(collection.name(), auth::Level::RW);
 
-  if (exec != nullptr) {
-    bool canModify = exec->canUseCollection(collection.name(), auth::Level::RW);
-
-    if ((exec->databaseAuthLevel() != auth::Level::RW || !canModify)) {
-      return TRI_ERROR_FORBIDDEN;
-    }
+  if ((exec.databaseAuthLevel() != auth::Level::RW || !canModify)) {
```

Comment:
IMHO we should switch the operands here, i.e.
```suggestion
  if (!canModify || exec.databaseAuthLevel() != auth::Level::RW) {
```